[Timenikhil] iP

Timenikhil opened the pull request:
**AVA**

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://en.wikipedia.org/wiki/Getting_Things_Done))

AVA frees your mind of having to remember things you need to do. It's, 
* text-based
* easy to learn
* **~~FAST~~ SUPER FAST** to use

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/releases/latest).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE** (for now...)!

**Features:**
* -[x]  Managing tasks
* -[ ] Managing deadlines (coming soon)
* -[ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice _Java_ too. Here's the `main` method:

```java
public class Main {
  public static void main(String[] args) {
    Application.launch(MainApp.class, args);
  }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Timenikhil committed:
Rename Chatbot
Make it Greet and Exit

Timenikhil committed:
replace concatenation with text block

Timenikhil committed:
rename Duke class to AVA
Add simple text repetition feature in chatbot.

Timenikhil committed:
fix redundant errors

Timenikhil committed:
create Task
create TaskManager
refactor AVA to support multiple instances

Timenikhil committed:
finish refactoring AVA to be more generic

Timenikhil committed:
finish list making response

Timenikhil committed:
finish list marking response

Timenikhil committed:
basic error handling and deleting

Timenikhil committed:
basic  deleting implemented

Timenikhil committed:
create different tasks

Timenikhil committed:
create enum error

Timenikhil committed:
attempt to do automated text testing

Timenikhil committed:
clean file structure

Timenikhil committed:
Improve TaskManager

Timenikhil committed:
Make tasks more detailed

Timenikhil committed:
improve task management

Timenikhil committed:
make AVA's greetings cuter to align with her personality

Timenikhil committed:
follow Coding standard

Timenikhil committed:
refactor code to follow Coding standard

Timenikhil committed:
Merge pull request #2 from Timenikhil/ErrorHandling

Handle errors better

Timenikhil committed:
create fileManager

Timenikhil committed:
move default taskManager to file reader for parity

Timenikhil committed:
fix bugs in FileManager.java

Timenikhil committed:
Merge pull request #3 from Timenikhil/master

Bring over FileManagement to Level 7 branch

Timenikhil committed:
add data to gitignore

Timenikhil committed:
add data to gitignore, add filereader

Timenikhil committed:
use bufferedreader

Timenikhil committed:
Merge pull request #4 from Timenikhil/branch-Level-7

Add Save to Disk ability to AVA

Timenikhil committed:
Improve JavaDoc

Timenikhil committed:
add basic time understanding

Timenikhil committed:
Merge pull request #5 from Timenikhil/branch-Level-8

Teach AVA to intepret date time

Timenikhil committed:
Merge pull request #6 from Timenikhil/add-gradle-support

Add gradle support

Timenikhil committed:
add basic test files

Timenikhil committed:
add basic searching to find tasks

Timenikhil committed:
Merge pull request #7 from Timenikhil/Find

add basic searching to find tasks

Timenikhil committed:
refactor code to add test root

Reallyeasy1 commented on .gitignore:
```diff
@@ -15,3 +15,5 @@ bin/
 /text-ui-test/ACTUAL.TXT
 text-ui-test/EXPECTED-UNIX.TXT
+
+/data
```
Good that you added /data in .gitignore

Reallyeasy1 commented on src/main/java/ava/AVA.java:
```diff
@@ -0,0 +1,119 @@
+        currentInput = s;
+    }
+
+    //todo:have a non printstream version
```
Is there are reason why the todo here is not capitalized?

Reallyeasy1 commented on src/main/java/ava/AVA.java:
```diff
@@ -0,0 +1,119 @@
+
+    /**
+     * Decides if AVA is running or not
+     * //TODO: switch to a state based system like Operating System Threads
```
Is there a reason why you put the TODO in the javadocs? Is there a reason why the TODO format is not consistent with the next TODO?

Reallyeasy1 commented on src/main/java/ava/AVA.java:
```diff
@@ -0,0 +1,119 @@
+    /**
+     * Prints AVA's response to given PrintStream
+     * @param out PrintStream to print AVA's response to
+     * //TODO:refactor mark and unmark to remove redundancy
```
Is there a reason why you put the TODO in the javadocs?

Reallyeasy1 commented on src/main/java/ava/files/FileManager.java:
```diff
@@ -0,0 +1,98 @@
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return new ArrayList<Task>(){
```
Is there a reason why you create an anonymous class to store the Tasks? Should this anonymous class be a seperate class by itself (ie. TaskList) for better readability?

Reallyeasy1 commented on src/main/java/ava/task/Tasks/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+        sb.append("Deadline: ");
+        sb.append(super.toString());
+        sb.append("[By :");
+        sb.append(time.format(DateTimeFormatter.ISO_DATE_TIME));
```
Good that you use ISO_DATE_TIME rather than the normal string format

Reallyeasy1 reviewed (commented):
Good job overall. Just need minor changes.

The0nlyJuan commented on src/main/java/ava/AVA.java:
```diff
@@ -0,0 +1,119 @@
+            out.println("Here are your tasks:");
+            out.println(list);
+        } else if(currentInput.startsWith("mark")){
+            int taskId;
```
try to intialise taskId insteal of leaving it

The0nlyJuan commented on src/main/java/ava/task/TaskManager.java:
```diff
@@ -0,0 +1,58 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskManager {
```
Could have added some javadoc for this

The0nlyJuan reviewed (commented):
Clean code overall, could have included more javadocs

Timenikhil committed:
add GUI to AVA

Timenikhil committed:
Merge pull request #8 from Timenikhil/GUI

add GUI to AVA

Timenikhil committed:
Merge pull request #9 from Timenikhil/branch-Level-10

Branch level 10

Timenikhil committed:
add varargs to errors

Timenikhil committed:
Merge pull request #10 from Timenikhil/Varargs

add varargs to errors

Timenikhil committed:
add assertions

Timenikhil committed:
Merge pull request #11 from Timenikhil/branch-A-Assertions

add assertions

Timenikhil committed:
improve code Quality

Timenikhil committed:
Merge pull request #12 from Timenikhil/branch-A-CodeQuality

improve code Quality

Timenikhil committed:
Create gradle.yml

Add CI to ensure the code is compiled and build with gradle on pushing to Github

Timenikhil committed:
Stream response to user to cut waiting time

The response is streamed to the caller using java streams to prevent buffering and notify the user of the result, as soon as it is done. This improves percieved latency and memory requirements.

Timenikhil committed:
Add priority scheduling

Allows user to set priority for their tasks

Timenikhil committed:
Merge pull request #13 from Timenikhil/branch-C-priority

Add priority scheduling

Timenikhil committed:
Fix code quality issues

Fix the code quality issues highlighted by the bot issue

Timenikhil committed:
Fix Header Comments

Fix header comments to follow java coding standard

Timenikhil committed:
Establish JUnit Test for Parser

set up testing framework successfully

Timenikhil committed:
Add todo , deadline execution

Timenikhil committed:
Fix Parser error message

Timenikhil committed:
Add tests to Parser for date times

Timenikhil committed:
passes textui test

Timenikhil committed:
Improve SlAP for main parser

Timenikhil committed:
Improve SLAP for main parser

Timenikhil committed:
Merge remote-tracking branch 'origin/improveFind' into improveFind

Timenikhil committed:
Improve Code Quality and Add serialization

Timenikhil committed:
Fix serialization deserialization

Timenikhil committed:
Fix files bug in which creating FileManager erases data file

Timenikhil committed:
Merge pull request #15 from Timenikhil/improveFind

Improve find

Timenikhil committed:
fix merge failure

Timenikhil committed:
Merge pull request #16 from Timenikhil/improveFind

improveFind

Timenikhil committed:
fix checkstyle

Timenikhil committed:
add JavaFX dependencies

Timenikhil committed:
fix JavaFX dependencies

Timenikhil committed:
add images for characters for AVA GUI

Timenikhil committed:
Fix Checkstyle issues

Timenikhil committed:
Fix GUI to work

Timenikhil committed:
add character images to git

Timenikhil committed:
improve GUI and remove bugs

Timenikhil committed:
add tests to test tasks

Timenikhil committed:
Merge pull request #17 from Timenikhil/BetterGUI

Improve GUI

Timenikhil committed:
fix checkstyle issues

Timenikhil committed:
Merge pull request #18 from Timenikhil/BetterGUI

BetterGUI

Timenikhil committed:
improve datetime formatting and add tests

Timenikhil committed:
update Readme to suit AVA

Timenikhil committed:
add icon.jpg

Timenikhil committed:
Update README.md

Timenikhil committed:
Update Command.java

Timenikhil committed:
Update Parser.java

Timenikhil committed:
Update AVA.java

Timenikhil committed:
Fix Github access issue

Timenikhil committed:
Merge pull request #19 from Timenikhil/C-help

C help

Timenikhil committed:
test if github is still broken

Timenikhil committed:
update user guide link

Timenikhil committed:
cite sources properly

Timenikhil committed:
reinstate cited assets

Timenikhil committed:
Create temp.txt

Timenikhil committed:
Add free assets to AVA

Timenikhil committed:
Merge remote-tracking branch 'origin/master'

Timenikhil committed:
fix files lost to github push failure

Timenikhil committed:
add Ui.png to documentation

Timenikhil committed:
add Ui picture to Readme

Timenikhil committed:
style Readme further

Timenikhil committed:
hide link in formatted text

Timenikhil committed:
Use cayman theme

Timenikhil committed:
update userGuide with screenshot and product description

Timenikhil committed:
show download links for AVA

Timenikhil committed:
Add Table of Contents to user Guide

Timenikhil committed:
Add Quick Start to User Guide

Timenikhil committed:
Add Steps to QuickStart for AVA UG

Timenikhil committed:
Experiment with code blocks on cayman

Timenikhil committed:
Complete Quick Start guide for AVA

Timenikhil committed:
Fix formatting errors in User Guide

Timenikhil committed:
Fix numbering issue

Timenikhil committed:
Escape quote by indenting code block

Timenikhil committed:
Change order of quote and code to see if it helps

Timenikhil committed:
Try github alert

Timenikhil committed:
Try out Jekyll Callout

Timenikhil committed:
Revert to normal quotation

Timenikhil committed:
Finish User Guide

Timenikhil committed:
Make error handling more robust

Timenikhil committed:
Display errors to user

Timenikhil committed:
Finalise features and ensure proper error handling
